src/System.Text.Json/tests/Utf8JsonReaderTests.TextEquals.cs

```diff
@@ -0,0 +1,447 @@
+            bool foundId = false;
+            bool foundTransports = false;
+            bool foundValue = false;
+            bool foundArrayValue = false;
+
+            string jsonString = "{\"conne\\u0063tionId\":\"123\",\"availableTransports\":[\"My name is \\\"Ahson\\\"\"]}";
+            byte[] utf8Data = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(utf8Data, isFinalBlock: true, state: default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.PropertyName)
+                {
+                    if (json.TextEquals(connectionId) && json.TextEquals("connectionId".AsSpan()))
+                    {
+                        foundId = true;
+                    }
+                    else if (json.TextEquals(availableTransports) && json.TextEquals("availableTransports".AsSpan()))
+                    {
+                        foundTransports = true;
+                    }
+                }
+                else if (json.TokenType == JsonTokenType.String)
+                {
+                    if (json.TextEquals(value123) && json.TextEquals("123".AsSpan()))
+                    {
+                        foundValue = true;
+                    }
+                    else if (json.TextEquals(embeddedQuotes) && json.TextEquals("My name is \"Ahson\"".AsSpan()))
+                    {
+                        foundArrayValue = true;
+                    }
+                }
+            }
+
+            Assert.True(foundId);
+            Assert.True(foundTransports);
+            Assert.True(foundValue);
+            Assert.True(foundArrayValue);
+        }
+
+        [Theory]
+        [InlineData("{\"name\": 1234}", "name", true)]
+        [InlineData("{\"name\": 1234}", "namee", false)]
+        [InlineData("{\"name\": 1234}", "na\\u006de", false)]
+        [InlineData("{\"name\": 1234}", "", false)]
+        [InlineData("{\"\": 1234}", "name", false)]
+        [InlineData("{\"\": 1234}", "na\\u006de", false)]
+        [InlineData("{\"\": 1234}", "", true)]
+        [InlineData("{\"na\\u006de\": 1234}", "name", true)]
+        [InlineData("{\"na\\u006de\": 1234}", "namee", false)]
+        [InlineData("{\"na\\u006de\": 1234}", "na\\u006de", false)]
+        [InlineData("{\"na\\u006de\": 1234}", "", false)]
+        public static void TestTextEquals(string jsonString, string lookUpString, bool expectedFound)
+        {
+            byte[] lookup = Encoding.UTF8.GetBytes(lookUpString);
+            byte[] utf8Data = Encoding.UTF8.GetBytes(jsonString);
+            bool found = false;
+
+            var json = new Utf8JsonReader(utf8Data, isFinalBlock: true, state: default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.PropertyName)
+                {
+                    if (json.TextEquals(lookup) && json.TextEquals(lookUpString.AsSpan()))
+                    {
+                        found = true;
```

Comment:
Since none of these loops ever set found back to false (within the loops), perhaps they should break after `found = true;`?